x-pack/plugins/fleet/cypress/integration/agent.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+const agent = {
+  id: 'agent-1',
+  access_api_key_id: 'abcdefghijklmn',
+  action_seq_no: [-1],
+  active: true,
+  agent: {
+    id: 'agent-1',
+    version: '8.1.0',
+  },
+  enrolled_at: '2022-03-07T14:02:00Z',
+  local_metadata: {
+    elastic: {
+      agent: {
+        'build.original': '8.1.0',
+        id: 'agent-1',
+        log_level: 'info',
+        snapshot: true,
+        upgradeable: true,
+        version: '8.1.0',
+      },
+    },
+    host: {
+      architecture: 'x86_64',
+      hostname: 'agent-1',
+      id: 'AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE',
+      ip: ['127.0.0.1/8'],
+      mac: ['ab:cd:12:34:56:78'],
+      name: 'agent-1',
+    },
+    os: {
+      family: 'darwin',
+      full: 'Mac OS X(10.16)',
+      kernel: '21.3.0',
+      name: 'Mac OS X',
+      platform: 'darwin',
+      version: '10.16',
+    },
+  },
+  policy_id: 'policy-1',
+  type: 'PERMANENT',
+  default_api_key: 'abcdefg',
+  default_api_key_id: 'abcd',
+  policy_output_permissions_hash: 'somehash',
+  updated_at: '2022-03-07T16:35:03Z',
+  last_checkin_status: 'online',
+  last_checkin: '2022-03-07T16:34:56Z',
+  policy_revision_idx: 1,
+  policy_coordinator_idx: 1,
+  policy_revision: 1,
+  status: 'online',
+  packages: [],
+};
+
+describe('View agents', () => {
+  beforeEach(() => {
+    cy.intercept('/api/fleet/agents/setup', { isReady: true });
```

Comment:
A hack to setup the fleet app, is this too much of a hack? there may be a more elegant way of doing this but I didn't spot it

Reply:
I think it is fine since we don't want to start a real fleet-server and agents